clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -3717,6 +3717,30 @@ def err_sme_definition_using_za_in_non_sme_target : Error<
   "function using ZA state requires 'sme'">;
 def err_sme_definition_using_zt0_in_non_sme2_target : Error<
   "function using ZT0 state requires 'sme2'">;
+def warn_sme_streaming_caller_pass_args_to_non_streaming : Warning<
+  "streaming caller passes a VL-dependent argument to non-streaming callee, "
+  "the streaming and non-streaming vector lengths may be different">,
+  InGroup<IgnoredAttributes>;
```

Comment:
I don't think 'IgnoredAttributes' is correct.

These diagnostics probably deserve to be in their own group.

Reply:
Because the attribute is not ignored.

Reply:
Add new warning group AArch64SMEAttributes.